From 8d429fc1e1ce68609abeb71b8bd2a858be85424a Mon Sep 17 00:00:00 2001
From: Vladislav Belogrudov <vladislav.belogrudov@oracle.com>
Date: Fri, 20 Oct 2017 14:58:48 +0300
Subject: [PATCH] Non-lvm iscsi backends should run iscsid on compute and storage nodes

Cinder backends that utilize iSCSI protocol require iscsid on
compute and storage nodes to handle volumes and their backups.

Change-Id: I1c735d2abace4b46879be99731ead47d578335ea
Closes-Bug: #1725258
---

diff --git a/ansible/roles/iscsi/tasks/start.yml b/ansible/roles/iscsi/tasks/start.yml
index c2934cf..82e6959 100644
--- a/ansible/roles/iscsi/tasks/start.yml
+++ b/ansible/roles/iscsi/tasks/start.yml
@@ -18,7 +18,7 @@
       - "/sys/kernel/config:/configfs"
       - "cinder:/var/lib/cinder"
       - "iscsi_info:/etc/iscsi"
-  when: ( ( inventory_hostname in groups['compute'] or inventory_hostname in groups['cinder-volume'] ) and enable_cinder_backend_lvm | bool )
+  when: ( ( inventory_hostname in groups['compute'] or inventory_hostname in groups['cinder-volume'] ) and enable_cinder_backend_iscsi | bool )
          or ( inventory_hostname in groups['ironic-conductor'] and enable_ironic | bool )
 
 - name: Starting tgtd container
